### PR TITLE
Grace is a WYSIWYG 2D plotting tool for the X Window System and M*tif.

### PR DESCRIPTION
**Pull request by NetForceExplorer**
###### Motivation for this change
Grace is a WYSIWYG 2D plotting tool for the X Window System and M*tif and not currently packaged for Nix. I intend to maintain this derivation and want to get more involved, hence adding myself as a maintainer.

This is my first contribution to an open-source project and I welcome every feedback.

###### Things done
	- maintainer-list.nix: add netforceexplorer (me)
      	- all-packages.nix: add grace
	- default.nix: build instructions for grace

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [X] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Artturin** [comment]
The maintainer list edit should be in a separate commit and before the init commit https://git-rebase.io/#reorder


Read contributing.md https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -7616,6 +7616,13 @@
     githubId = 34162313;
     name = "Jason Wing";
   };
+  netforceexplorer = {
+    email = "netforce-explorer-dev@outlook.com";
+    github = "netforceexplorer";
+    githubId = 20754345;
+    name = "Tobias Fischbach";
+
```
Comment (Artturin) [inline maintainers/maintainer-list.nix:7624]:
```suggestion
```

File: pkgs/tools/graphics/grace/default.nix [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+, libX11
+, libSM
+, libXext
+, libXmu
+, libXp
+, libXpm
+, libXt
+}:
+
+gccStdenv.mkDerivation rec {
```
Comment (Artturin) [inline pkgs/tools/graphics/grace/default.nix:19]:
```suggestion
stdenv.mkDerivation rec {
```

File: pkgs/tools/graphics/grace/default.nix [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      libX11
+      libSM
+      libXext
+      libXmu
+      libXp
+      libXpm
+      libXt
+    ];
+
+    configurePhase = ''
```
Comment (Artturin) [inline pkgs/tools/graphics/grace/default.nix:43]: Use configureFlags, you can find examples by searching this repo with ripgrep

File: pkgs/tools/graphics/grace/default.nix [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        ./configure --enable-grace-home=$out \
+                    --prefix=$out \
+                    --disable-debug \
+                    --with-bundled-t1lib=yes
+    '';
+
+    buildPhase = ''
+        NIX_CFLAGS_COMPILE="-Wno-implicit-function-declaration -Wno-format-security $NIX_CFLAGS_COMPILE"
```
Comment (Artturin) [inline pkgs/tools/graphics/grace/default.nix:51]: Why is this needed. If they're really needed then this isnt the correct way to declare them

File: pkgs/tools/graphics/grace/default.nix [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    ];
+
+    configurePhase = ''
+        ./configure --enable-grace-home=$out \
+                    --prefix=$out \
+                    --disable-debug \
+                    --with-bundled-t1lib=yes
+    '';
+
+    buildPhase = ''
```
Comment (Artturin) [inline pkgs/tools/graphics/grace/default.nix:50]: Overriding buildPhase is not necessary

File: pkgs/tools/graphics/grace/default.nix [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+        make -j $(nproc)
+        make install
+    '';
+
+    meta = with lib; {
+      homepage = "https://plasma-gate.weizmann.ac.il/Grace/";
+      description = "Grace is a WYSIWYG 2D plotting tool for the X Window System and M*tif. Grace is a descendant of ACE/gr, also known as Xmgr, and runs on practically any version of Unix-like OS.";
+    platforms = platforms.linux ++ platforms.darwin;
+    license = {
```
Comment (Artturin) [inline pkgs/tools/graphics/grace/default.nix:61]:
```suggestion
    license = licenses.gpl2Only;
```

**Review by Artturin — CHANGES_REQUESTED** [review]

File: pkgs/tools/graphics/grace/default.nix (hunk 2 above)
Comment (NetForceExplorer) [reply pkgs/tools/graphics/grace/default.nix:19]: Hi,

I used gccStdenv as clang failed to built but is the default on Darwin systems.

File: pkgs/tools/graphics/grace/default.nix (hunk 4 above)
Comment (NetForceExplorer) [reply pkgs/tools/graphics/grace/default.nix:51]: -Wno-implicit-function-declaration is indeed not necessary.

-Wno-format-security is required for a successful built. Grace does not implement format strings correctly. Example error message:

objects.c:960:22: error: format string is not a string literal (potentially insecure) [-Werror,-Wformat-security]

File: pkgs/tools/graphics/grace/default.nix [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+      homepage = "https://plasma-gate.weizmann.ac.il/Grace/";
+      description = "Grace is a WYSIWYG 2D plotting tool for the X Window System and M*tif. Grace is a descendant of ACE/gr, also known as Xmgr, and runs on practically any version of Unix-like OS.";
+    platforms = platforms.linux ++ platforms.darwin;
+    license = licenses.gpl2Only;
+      # GNU GENERAL PUBLIC LICENSE
+
+      # Version 2, June 1991
+
+      # Copyright (C) 1989, 1991 Free Software Foundation, Inc.
+      # 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
```
Comment (Artturin) [inline pkgs/tools/graphics/grace/default.nix:67]: These comments need to be removed too

**Comment by Artturin** [comment]
To keep the commit history clean combine use rebase with fixup https://git-rebase.io/#fixup

File: pkgs/tools/graphics/grace/default.nix [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+, libSM
+, libXext
+, libXmu
+, libXp
+, libXpm
+, libXt
+}:
+
+stdenv.mkDerivation rec {
+    pname = "grace";
```
Comment (SuperSandro2000) [inline pkgs/tools/graphics/grace/default.nix:20]: Please format this file with nixpkgs-fmt.

File: pkgs/tools/graphics/grace/default.nix [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+    version = "5.1.25";
+    src = fetchurl {
+        url = "ftp://ftp.fu-berlin.de/unix/graphics/grace/src/grace5/grace-5.1.25.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/tools/graphics/grace/default.nix:23]:
```suggestion
        url = "ftp://ftp.fu-berlin.de/unix/graphics/grace/src/grace5/grace-${version}.tar.gz";
```

File: pkgs/tools/graphics/grace/default.nix [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+    buildPhase = ''
+        NIX_CFLAGS_COMPILE="-Wno-implicit-function-declaration -Wno-format-security $NIX_CFLAGS_COMPILE"
+
+        make -j $(nproc)
+        make install
+    '';
+
+    meta = with lib; {
+      homepage = "https://plasma-gate.weizmann.ac.il/Grace/";
+      description = "Grace is a WYSIWYG 2D plotting tool for the X Window System and M*tif. Grace is a descendant of ACE/gr, also known as Xmgr, and runs on practically any version of Unix-like OS.";
```
Comment (SuperSandro2000) [inline pkgs/tools/graphics/grace/default.nix:59]: This is to long for the short description. Use longDescription for that and add a shorter one here.

File: pkgs/tools/graphics/grace/default.nix (hunk 4 above)
Comment (NetForceExplorer) [reply pkgs/tools/graphics/grace/default.nix:51]: Performing further error checking, we can ignore -Wno-implicit-function-declaration. The crashing binary is only produced if clang is used.  But the -Wno-format-security is still necessary for a successful build due to the format strings. I deleted the build phase overwrite but what is the preferred way of setting of setting cflags?

Now I use:
`NIX_CFLAGS_COMPILE=-Wno-format-security `

Thank you for your patience.

Comment (Artturin) [reply pkgs/tools/graphics/grace/default.nix:51]:
```
NIX_CFLAGS_COMPILE = [ "-Wno-format-security" ] 
  ++ lib.optionals stdenv.cc.isClang
    [ "-Wno-implicit-function-declaration" ];
```

Don't put this in any phase

**Comment by Artturin** [comment]
Drop the `Delete maintainer-list.nix` commit with rebase

**Comment by Artturin** [comment]
Read

File: pkgs/tools/graphics/grace/default.nix [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+    libXp
+    libXpm
+    libXt
+  ];
+
+  enableParallelBuilding = true;
+
+  NIX_CFLAGS_COMPILE = [ "-Wno-format-security" ]
+    ++ lib.optionals stdenv.cc.isClang
+    [ "-Wno-implicit-function-declaration" "-O1" ];
```
Comment (SuperSandro2000) [inline pkgs/tools/graphics/grace/default.nix:47]:
```suggestion
    ++ lib.optionals stdenv.cc.isClang [ "-Wno-implicit-function-declaration" "-O1" ];
```

File: pkgs/tools/graphics/grace/default.nix [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+
+  configureFlags = [
+    "--disable-debug"
+    "--with-bundled-t1lib=yes"
+  ];
+
+  preConfigure = ''
+    configureFlags="$configureFlags --enable-grace-home=$out"
+  '';
```
Comment (SuperSandro2000) [inline pkgs/tools/graphics/grace/default.nix:56]:
```suggestion
  configureFlags = [ "--enable-grace-home=$out" ];
```

File: pkgs/tools/graphics/grace/default.nix [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+  meta = with lib; {
+    homepage = "https://plasma-gate.weizmann.ac.il/Grace/";
+
+    description = "Grace is a WYSIWYG 2D plotting tool.";
+    longDescription = "Grace is a WYSIWYG 2D plotting tool for the X Window System and M*tif. Grace is a descendant of ACE/gr, also known as Xmgr, and runs on practically any version of Unix-like OS.";
+
+    platforms = platforms.linux ++ platforms.darwin;
+    license = licenses.gpl2Only;
+
+    maintainers = with maintainers; [ netforceexplorer ];
```
Comment (SuperSandro2000) [inline pkgs/tools/graphics/grace/default.nix:67]:
```suggestion
    homepage = "https://plasma-gate.weizmann.ac.il/Grace/";
    description = "Grace is a WYSIWYG 2D plotting tool.";
    longDescription = "Grace is a WYSIWYG 2D plotting tool for the X Window System and M*tif. Grace is a descendant of ACE/gr, also known as Xmgr, and runs on practically any version of Unix-like OS.";
    platforms = platforms.unix;
    license = licenses.gpl2Only;
    maintainers = with maintainers; [ netforceexplorer ];
```

File: pkgs/tools/graphics/grace/default.nix (hunk 12 above)
Comment (NetForceExplorer) [reply pkgs/tools/graphics/grace/default.nix:56]: Hi,

I used this preconfigure hook as a solution for a specific problem. The variable $out is not evaluated if it's in the configureFlags (of course I use double-quotes). If I add $out in preConfigure it works and is evaluated. Does the usage of preConfigure impose problems?

**Comment by Artturin** [comment]
> Drop the `Delete maintainer-list.nix` commit with rebase



> Read https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md



> To keep the commit history clean use rebase with fixup https://git-rebase.io/#fixup

File: pkgs/tools/graphics/grace/default.nix [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+  configureFlags = [
+    "--disable-debug"
+    "--with-bundled-t1lib=yes"
+  ];
+
+  meta = with lib; {
+    homepage = "https://plasma-gate.weizmann.ac.il/Grace/";
+    description = "Grace is a WYSIWYG 2D plotting tool.";
+    longDescription = "Grace is a WYSIWYG 2D plotting tool for the X Window System and M*tif. Grace is a descendant of ACE/gr, also known as Xmgr, and runs on practically any version of Unix-like OS.";
+    platforms = platforms.linux ++ platforms.darwin;
```
Comment (SuperSandro2000) [inline pkgs/tools/graphics/grace/default.nix:59]:
```suggestion
    platforms = platforms.unix;
```

File: pkgs/tools/graphics/grace/default.nix [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+
+  NIX_CFLAGS_COMPILE = [ "-Wno-format-security" ]
+    ++ lib.optionals stdenv.cc.isClang [ "-Wno-implicit-function-declaration" "-O1" ];
+
+  preConfigure = ''configureFlags="$configureFlags --enable-grace-home=$out"'';
+
+  configureFlags = [
+    "--disable-debug"
+    "--with-bundled-t1lib=yes"
+  ];
```
Comment (SuperSandro2000) [inline pkgs/tools/graphics/grace/default.nix:53]:
```suggestion
  configureFlags = [
    "--disable-debug"
    "--enable-grace-home=${placeholder "out"}"
    "--with-bundled-t1lib=yes"
  ];
```

**Comment by AndersonTorres** [comment]
Closing as dead
